js/NewsChannelOperations.js

```diff
@@ -0,0 +1,44 @@
+function SaveEmail(){
+  var emailAddresss= document.getElementById('txtEmailAddress').value;
+  if (ValidateEmail(emailAddresss)){
+    localStorage.setItem('email',emailAddresss);
+  }
+}
+function ValidateEmail(mail)
+{
+  if (/^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/.test(mail))
+  {
+    return true;
+  }
+}
+function ShowHideDiv(selectedElement) {
+  if(selectedElement.value> 0){
+    for(var index =1 ;index< newsChannels.length; index++){
+
+      if (selectedElement.value == index){
```

Comment:
instead of checking this condition everytime. first hide everything and show only the one selected. This way you can reduce the depth and improve performance.

Reply:
i am using same div (list of news info displayed in div on page load) to show as popup by changing class name it is not possible to hide everything.